script/ts/multisig-cli.ts

```diff
@@ -0,0 +1,543 @@
+      }
+
+    } catch (error) {
+      console.log("❌ Simulation request failed:", (error as Error).message);
+      console.log("\n⚠️  Proceeding without simulation...");
+      return false;
+    }
+  }
+
+  /**
+   * Handle errors with appropriate logging and context
+   */
+  private handleError(message: string, error: unknown): void {
+    console.error(`\n❌ ${message}`);
+    
+    if (error instanceof MultisigCliError) {
+      console.error(`Code: ${error.code}`);
+      console.error(`Details: ${error.message}`);
+      if (error.cause) {
+        console.error(`Cause: ${error.cause.message}`);
+      }
+    } else if (error instanceof Error) {
+      console.error(`Details: ${error.message}`);
+    } else {
+      console.error(`Details: ${String(error)}`);
+    }
+    
+    console.error("\n💡 For help, run: npm run multisig-cli --help");
+    process.exit(1);
+  }
+}
+
+// ============================================================================
+// CLI Argument Parsing and Main Execution
+// ============================================================================
+
+/**
+ * Parse command line arguments into typed options
+ */
+function parseCliArgs(): CliOptions {
+  const args = process.argv.slice(2);
+  const options: CliOptions = {};
+
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+
+    switch (arg) {
+      case "--help":
+      case "-h":
+        return { help: true };
+        
+      case "--simulate-only":
+        options.simulateOnly = true;
+        break;
+        
+      case "--transfer-ownership":
+        options.transferOwnership = true;
+        break;
+        
+      case "--accept-ownership":
+        options.acceptOwnership = true;
+        break;
+        
+      case "--amount":
+      case "--eth":
+      case "-a":
+        const ethValue = args[i + 1];
+        if (ethValue && !ethValue.startsWith("-")) {
+          options.ethAmount = ethValue;
+          i++;
+        }
+        break;
+        
+      case "--hlg":
+      case "--direct-hlg":
+        const hlgValue = args[i + 1];
+        if (hlgValue && !hlgValue.startsWith("-")) {
+          options.hlgAmount = hlgValue;
+          i++;
+        }
+        break;
+        
+      default:
+        // Positional argument - treat as ETH amount
+        if (arg && !arg.startsWith("-")) {
+          options.ethAmount = arg;
+        }
+        break;
+    }
+  }
+
+  return options;
+}
+
+/**
+ * Main CLI execution function
+ */
+async function main(): Promise<void> {
+  try {
+    const options = parseCliArgs();
+    const cli = new MultisigCLI();
+
+    // Handle special operations first
+    if (options.help) {
+      cli.displayHelp();
+      return;
+    }
+
+    if (options.transferOwnership) {
+      await cli.transferStakingRewardsOwnership();
+      return;
+    }
+
+    if (options.acceptOwnership) {
+      await cli.generateAcceptOwnershipTransaction();
+      return;
+    }
+
+    // Handle main operations
```

Comment:
Ok i think if we have different operations from the CLI, we need to have a command with their own sets of flags. Its really confusing to have 2 operations that do different things look the same. 


```
npm run multisig-cli batch --param1  --param2
npm run multisig-cli deposit --param1  --param2
npm run multisig-cli accept-ownership --param1  --param2
npm run multisig-cli transfer-ownership --param1  --param2
```

the above is much clearer than before. The current one requires the caller to know which flag will do which operation.

Reply:
updated this stuff too